src/daemon/dbus_service.cpp

```diff
@@ -41,15 +48,19 @@ namespace TemplateDBusService::Daemon
         if (connection_id_ == 0)
```

Comment:
consider change to
`if (0 == connection_id)
`

(General comment for all files, always have the CONSTANT first, not the variable))

Reply:
So... I personally do not like this style (I think it reads less naturally... if you are not Yoda ;) ... it is called Yoda style for a reason :) ...) and have never worked in a code base that uses it (if that counts for something :).

GCC, Clang and clang-tidy also all warn if you assign by accident.

Reply:
I would say that it is also not that common. As I said, never used it at work but also, not a single one of the ~40 FOSS projects (GNOME, freedesktop.org, Chromium, LLVM, GCC, Qt, rsync, etc.) I have available locally at the moment use it. And I can not recall ever seeing a specific project that does use it. (Not saying there are none, just saying I think most of the world does not speak Yoda. :)

Remnant from environments/compilers were it can not be detected?

Reply:
It was part of MISRA C in the past so I'm personally used to it :) since we had to write MISRA C compliant software (many years ago though). Anyway, I agree, it looks like the modern guidleines, even AUTOSAR C++14 which is based on MISRA actually allows if (VAR == CONSTANT) now.
So, keep on what you are used to :).

Reply:
OK. :) But also... (talked a bit about this in person but writing it down here for others...)

I think it is better to rely on "tools" to catch errors that this convention is supposed to prevent. Not having a tool to detect that the literal is on the left side of `==` requires manual review to make sure that the convention is followed.

Could of course e.g. write a `clang-tidy` check for this, but since `clang-tidy` (and GCC and Clang) already detects that a `=` is used in an if-expression... we already have tools that prevent errors that Yoda conditionals is meant to prevent. :)